src/main/java/com/shop/service/UserServiceImpl.java

```diff
@@ -26,22 +28,28 @@ public UserServiceImpl(UserRepository userRepository, RoleRepository roleReposit
         this.passwordEncoder = passwordEncoder;
     }
 
-    @Override
-    public User createUserFromForm(RegistrationDto registrationDto) {
-        return new User(Calendar.getInstance(), registrationDto.getUsername(), registrationDto.getPassword(), registrationDto.getEmail());
+    public User createUserFromRegistrationDto(RegistrationDto registrationDto) {
+        User user = new User(Calendar.getInstance(), registrationDto.getUsername().trim(),
+                registrationDto.getPassword().trim(), registrationDto.getEmail().trim());
+        Address address = createAddressFromAddressDto(user, registrationDto.getAddressDto());
+        user.getAddresses().add(address);
+        return user;
+    }
+
+    private Address createAddressFromAddressDto(User user, AddressDto addressDto){
+        return new Address(user, addressDto.getFirstname().trim(), addressDto.getLastname().trim(), addressDto.getCity().trim(),
+                addressDto.getStreet().trim(), addressDto.getZipCode().trim(), addressDto.getPhone().trim());
     }
 
     @Override
     public User register(User user) {
         user.setEnabled(true);
-        user.setRegistrationDate(Calendar.getInstance());
```

Comment:
Why did u remove this line?

Reply:
We dont need this, because it's in User constructor